DOC: document actions (#16)
pyansys/actions: doc/source/style-actions/index.rst

Style actions
=============

> The style actions allow for checking code and documentation quality compliance

Reviewer: ```suggestion
Style actions check code and documentation quality compliance
```

with PyAnsys guidelines.

> For using these actions, a project needs to use `pre-commit

Reviewer: ```suggestion
To use these actions, a project must use `pre-commit
```

> <https://pre-commit.com>`_ and `Vale <https://vale.sh>`_. If your project needs
> to implement these tools, please contact `pyansys.support@ansys.com

Reviewer: ```suggestion
<https://pre-commit.com>`_ and `Vale <https://vale.sh>`_. For help
implementing these tools, send an email to `pyansys.support@ansys.com
```

<mailto:pyansys.support@ansys.com>`_.

Code style action
-----------------
This action evaluates the code quality of your project by using `pre-commit`_. It is assumed that your project contains a

> ``.pre-comit-config.yaml`` file in the base root directory.

Reviewer 2: ```suggestion
``.pre-commit-config.yaml`` file in the root directory.
```

.. jinja:: code-style

    {{ inputs_table }}

> Code sample for using this action:

Reviewer: ```suggestion
Here is a code sample for using this action:
```

    code-style:
      name: Code style
      runs-on: ubuntu-latest
      steps:
        - name: "Run PyAnsys code style checks"
          uses: pyansys/actions/code-style@main

Doc style action
----------------
This action evaluates the documentation quality of your project by using
`Vale`_. It assumes that Vale's configuration file is stored in
``doc/.vale.ini``. A token is expected as input for Vale to indicate quality
errors by making comments. This token can be the ``${{ secrets.GITHUB_TOKEN }}``
one.

+--------------+--------------------------------------+-----------+---------+------------------+
| Input        | Description                          | Required  | Type    | Default          |
+==============+======================================+===========+=========+==================+
| vale-config  | Path to the Vale configuration file  | False     | string  | 'doc/.vale.ini'  |
+--------------+--------------------------------------+-----------+---------+------------------+
| token        | Required token for Vale commenter    | True      |         |                  |
+--------------+--------------------------------------+-----------+---------+------------------+

> Code sample for using this action:

Reviewer: ```suggestion
Here is a code sample for using this action:
```
